### PR TITLE
Problem bug report

### PR DESCRIPTION
**Pull request by geniusgordon**
fix problem limit in contest
hide error tolerance
add partial judge header
problem status link

### CODE REVIEW

File: problem/templates/problem/problem.html [hunk 1]
```diff
@@ -15,10 +15,10 @@
   </div>
   <div class="right">
     <div class="problem-option">
-      <a>Status</a> &nbsp;|&nbsp;
-      <a>Printable </a> &nbsp;|&nbsp;
-      <a>Rank</a> &nbsp;|&nbsp;
-      <a data-toggle='modal' data-target='#myModal'>Limits</a>
+	    <a href="/status/?pid={{ problem.pk }}">Status</a> &nbsp;|&nbsp;
```
Comment (nickelcheng) [inline problem/templates/problem/problem.html]: Please take care of indent

Comment (geniusgordon) [inline problem/templates/problem/problem.html]: fixed

**Comment by nickelcheng** [comment]
Can you add a choice for partial judge / special judge language? (C or CPP)
Use judge_language in problem model.

**Comment by nickelcheng** [comment]
When partial judge is chosen, partial judge code / header and special judge code must not be empty.

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
We also need a checking for special judge code.

**Comment by bruce3557** [comment]
Fine with this

**Comment by nickelcheng** [comment]
1. You didn't get correct judge type corresponding to judge source
screenshot from 2015-05-01 01 45 52
screenshot from 2015-05-01 01 46 47
2. partial judge header and other judge idis everywhere (it never disappears, no matter what judge type is chosen.)
screenshot from 2015-05-01 02 03 49
3. local judge got empty judge type by default
screenshot from 2015-05-01 02 02 27
So many obvious errors. Please test before you push code QQ

**Comment by nickelcheng** [comment]
It seems that my browser is push out from a group QQQ

**Comment by nickelcheng** [comment]
You can use forms.py to sieve options in next release. merge this PR now.

**Pull request merged**